src/locale/localed-util.c

```diff
@@ -351,44 +505,41 @@ static int read_next_mapping(const char* filename,
 
                 length = strv_length(b);
                 if (length < min_fields || length > max_fields) {
-                        log_error("Invalid line %s:%u, ignoring.", filename, *n);
-                        strv_free(b);
+                        log_warning("Invalid line %s:%u, ignoring.", strna(filename), *n);
                         continue;
 
                 }
 
-                *a = b;
+                *ret = TAKE_PTR(b);
                 return 1;
         }
 
+        *ret = NULL;
         return 0;
 }
 
 int vconsole_convert_to_x11(Context *c) {
-        const char *map;
-        int modified = -1;
+        int r, modified = -1;
+        X11Context *xc;
 
-        map = systemd_kbd_model_map();
+        assert(c);
 
-        if (isempty(c->vc_keymap)) {
-                modified =
-                        !isempty(c->x11_layout) ||
-                        !isempty(c->x11_model) ||
-                        !isempty(c->x11_variant) ||
-                        !isempty(c->x11_options);
+        xc = context_get_x11_context_safe(c);
```

Comment:
I find the logic in this function very hard to follow. `context_get_x11_context_safe()` returns one of the contexts depending on their contents, and then we operate on it, even though the function is supposed to convert vconsole to x11. Dunno, maybe add an overview comment of what the logic is?

Reply:
Sure.